bundles/framework/mydata/handler/PublishedMapsHandler.js

```diff
@@ -0,0 +1,292 @@
+                    if (layerParams !== '') {
+                        layerParams += ',';
+                    }
+                    layerParams += layer.id + '+' + layer.opacity;
+                    if (layer.style) {
+                        layerParams += '+' + layer.style;
+                    } else {
+                        layerParams += '+';
+                    }
+                });
+                mapStateParams += layerParams;
+            }
+        }
+        return mapStateParams;
+    }
+
+    createProjectionChangeDialog (cb) {
+        const dialog = Oskari.clazz.create('Oskari.userinterface.component.Popup');
+        const btn = dialog.createCloseButton(this.loc('projectionError').ok);
+        const cancel = dialog.createCloseButton(this.loc('projectionError').cancel);
+        cancel.setHandler(() => {
+            dialog.close(true);
+        });
+        btn.addClass('primary');
+        btn.setHandler(() => {
+            cb();
+            dialog.close(true);
+        });
+        dialog.show(this.loc('projectionError').title, this.loc('projectionError').msg, [cancel, btn]);
+        dialog.makeDraggable();
+    }
+
+    editView (data) {
+        const sandbox = Oskari.getSandbox();
+        const srs = data.state.mapfull.state.srs;
+        const embeddedMapUuid = data.uuid;
+        const supported = this.isCurrentProjectionSupportedForView(data);
+        if (!supported) {
+            this.createProjectionChangeDialog(() => {
+                window.location.href = this.constructUrlWithUuid(srs, embeddedMapUuid, data);
+            });
+        }
+
+        const resp = this.viewService.isViewLayersLoaded(data, sandbox);
+        if (resp.status) {
+            this.editRequestSender(data);
+        } else {
+            this.confirmSetState(() => {
+                this.editRequestSender(data);
+            }, resp.msg === 'missing');
+        }
+    }
+
+    setPublished (data) {
+        const view = this.getViewById(data.id);
+        if (view) {
+            const newState = !view.isPublic;
+            this.viewService.makeViewPublic(
+                data.id,
+                newState,
+                (isSuccess) => {
+                    if (isSuccess) {
+                        this.refreshViewsList();
+                    } else {
+                        this.showErrorMessage(
+                            this.loc('tabs.publishedmaps.error.makePrivate')
+                        );
+                    }
+                });
+        }
+    }
+
+    showOnMap (data) {
+        const srs = data.state.mapfull.state.srs;
+        const supported = this.isCurrentProjectionSupportedForView(data);
+        if (!supported) {
+            this.createProjectionChangeDialog(() => {
+                window.location.href = this.constructUrlWithUuid(srs, null, data);
+            });
+        }
+
+        this.setMapState(data, false, () => {
+            this.setMapState(data, true);
+        });
```

Comment:
Should this be in else?

```suggestion
        } else {
            this.setMapState(data, false, () => {
                this.setMapState(data, true);
            });
        }
```